core/tests/behavior/read_only.rs

```diff
@@ -44,6 +44,9 @@ pub fn behavior_read_only_tests(op: &Operator) -> Vec<Trial> {
         test_read_only_reader_with_range,
         test_read_only_reader_from,
         test_read_only_reader_tail,
+        test_read_only_reader_with_range_and_buffer,
```

Comment:
`read_only` tests are for services that read_only like http. Please add tests in `write` suite intead to cover s3 and fs.

Reply:
We should refactor the test suite to make it more contributor-friendly. I will propose my idea later after the holiday :love_letter: